include/deal.II/base/polynomials_integrated_legendre_sz.h

```diff
@@ -0,0 +1,92 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2015 - 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__polynomials_integrated_legendre_sz_h
+#define dealii__polynomials_integrated_legendre_sz_h
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/polynomial.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/thread_management.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * Class implementing the integrated Legendre polynomials described in the PhD thesis of Sabine Zaglmayer.
+ *
+ * This class was written based upon the existing deal.II Legendre class as a base, but with the coefficents adjusted
+ * so that the recursive formula is for the integrated Legendre polynomials described in the PhD thesis of
+ * Sabine Zaglmayer. The polynomials can be generated recursively from:
```

Comment:
Sorry, I had misread the part you asked me about. The 80 lines are for comments, yes, and we try to be within that margin also for the implementation, but we are not that strict.